sound/soc/sof/intel/hda-loader.c

```diff
@@ -296,8 +296,8 @@ int hda_dsp_cl_boot_firmware(struct snd_sof_dev *sdev)
 
 	chip_info = desc->chip_info;
 
-	stripped_firmware.data = plat_data->fw->data;
-	stripped_firmware.size = plat_data->fw->size;
+	stripped_firmware.data = plat_data->fw->data + plat_data->fw_offset;
+	stripped_firmware.size = plat_data->fw->size - plat_data->fw_offset;
```

Comment:
we should test if the offset is lower than the size, otherwise size_t is an unsigned and we will wrap-around

Reply:
added